### PR TITLE
Connect to api

### PR DESCRIPTION
**Pull request by elisha2kyakpo1**
In this branch, I did the following tasks;

1. Created an app id.
2. connect to the API.
3. Implemented add and remove.
4. Created a fetch data function to fetch data from API.
5. Did basic styling

### Note to the reviewer:
**You have to refresh to see the change after pressing the remove button. due to limitted time, I'm still working on an instant refresh after deleting a book. though this is not a requirement**

### CODE REVIEW

File: src/components/ApiData.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      .then((text) => text === success && dispatch(removeBook(id)))
+      .catch((error) => error === msg);
+  };
+};
+
+export {
+  removeBookApi,
+  addBookApi,
+  fetchBooksApi,
+};
```
Comment (DeliceLydia) [inline src/components/ApiData.js]: - [x] `remove` book feature should work the same as before even after the refactoring to use the API has been done currently `remove` is not working please fix that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @elisha2kyakpo1, 

Good job so far! However, there are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/ApiData.js (hunk 1 above)
Comment (elisha2kyakpo1) [reply src/components/ApiData.js]: ### Note to the reviewer:
**You have to refresh to see the change after pressing the remove button.**

File: src/components/ApiData.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      method: 'DELETE',
+      body: JSON.stringify({ item_id: id }),
+      headers: {
+        'content-type': 'application/json',
+      },
+    }).then((response) => (response.text()))
+      .then((text) => text === success && dispatch(removeBook(id)))
+      .catch((error) => error);
+  };
+};
```
Comment (MahmoudBakr23) [inline src/components/ApiData.js]: - [x] Refreshing the page when removing a book to re-render the data from the Redux store isn't a functional way to do so. Instead, you should automatically update the Redux store with the new array of books and display them after a book is removed! 🙏

File: src/components/ApiData.js (hunk 1 above)
Comment (MahmoudBakr23) [reply src/components/ApiData.js]: ### Hi @elisha2kyakpo1 😄 

Refreshing the page when removing a book to re-render the data from the Redux store isn't a functional way to do so. Instead, you should automatically update the Redux store with the new array of books and display them after a book is removed! 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @elisha2kyakpo1 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [x] Please attach an image of your great app to the README file 🙏 

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
Hi,

Good job so far! 👏🏼 @elisha2kyakpo1 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- You have a descriptive Readme file:heavy_check_mark:
- You have connected to API successfully:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous reviewers.
 
 #### Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**